### PR TITLE
Added statusObservers to Session

### PR DESCRIPTION
**Pull request by stephanheilner**

### CODE REVIEW

File: LDSAnnotations/Session.swift [hunk 1]
```diff
@@ -22,12 +22,15 @@
...
 import Foundation
 import PSOperations
+import Swiftification
 
 /// Communicates with the annotation service.
 ///
 /// Instances are lightweight; construct a new instance whenever the user's credentials change.
 public class Session: NSObject {
     
+    public let statusObservers = ObserverSet<Session>()
```
Comment (nickmshelley) [inline LDSAnnotations/Session.swift]: Why not just send status?

**Comment by nickmshelley** [comment]
.

File: LDSAnnotationsTests/SessionNotificationsTests.swift [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        session.authenticate { error in
+            XCTAssertNil(error)
+            
+            XCTAssertEqual(session.status, Session.Status.AuthenticationSuccessful)
+            expectation.fulfill()
+        }
+        
+        waitForExpectationsWithTimeout(30, handler: nil)
+        
+        XCTAssertEqual(statuses, [Session.Status.AuthenticationInProgress, Session.Status.AuthenticationSuccessful])
```
Comment (hiltonc) [inline LDSAnnotationsTests/SessionNotificationsTests.swift]: Nice, I like this!

**Comment by hiltonc** [comment]
👍

**Pull request merged**